php/class-asset-manager.php

```diff
@@ -435,10 +435,14 @@ public function add_core_dependencies( $asset ) {
 	 * @return void
 	 */
 	public function add_core_asset( $handle, $load_method = 'sync' ) {
-		global ${$this->core_assets_global};
-		$core_assets_ref    = ${$this->core_assets_global}->registered;
-		$in_footer          = isset( ${$this->core_assets_global}->in_footer ) ? ${$this->core_assets_global}->in_footer : [];
-		$core_asset_handles = array_keys( ${$this->core_assets_global}->registered );
+		if ( ! is_string( $this->core_assets_global ) || empty( $this->core_assets_global ) ) {
```

Comment:
The string check might be unnecessary here, but not bad to be extra safe since there isn't strict type checking on the property itself.